python/tvm/relay/frontend/tensorflow.py

```diff
@@ -436,6 +436,24 @@ def _impl(inputs, attr, params):
         return AttrCvt(op_name="copy", ignores=['message'])(inputs, attr)
     return _impl
 
+def _assert():
+    # ToDo: In general people want asserts to be gone from TensorFlow graphs
+    # when they are optimizing them, so converting it to a no-op is
+    # reasonable. However, it would be nice to have the option to keep them
+    # once Relay gets a Halt or Assert op.
+    return _no_op()
+
+def _no_op():
+    def _impl(inputs, attr, params):
```

Comment:
Trying to understand the semantics of a no-op? Can there be the TF graph

op1 --> NoOp --> op2

Where NoOp basically means connecting op1 and op2.

Reply:
That is possible with control dependencies. The reason no_op can appear when you call tf.assert() is that it is used as a control dependency. Quite unnecessarily, but it's there.

Reply:
In that case, will this implementation make sense?

~~~
def _impl(inputs, attr, params):
    if inputs:
      return inputs[0]
    return relay.const(0)
~~~

Reply:
My assumption here is that inputs are the value inputs, not control edges. If that understanding is correct, then there would never be inputs to no_op, since no_op doesn't have any operands and doesn't yield a value in TF. I would think that this part of the code is not concerned with translating control edges, which are a concern that can apply to many Graphdef ops and isn't a specific concern for a specific op.

Reply:
Understood. I have also seen similar behavior with TF no-op where it doesn't have any input operands. So, current implementation should be enough.

Reply:
As background, you can see this situation in this Tensorboard screenshot:

  https://pasteboard.co/IDcKMM4.png

Which is the graph generated from the second assert test in this PR.